### DISCUSSION TITLE
Allow wildcards in force:source:retrieve/deploy

### DISCUSSION THREAD

**Discussion by catalandres** [forcedotcom/cli#2363] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am trying to download only the classes related to a project in an org. We have named them consistently with a prefix and we would like to be able to retrieve and deploy only a subset of them. That way we will not risk overwriting another team's progress by deploying a past version of a file unrelated to our project.

**What are you trying to do**
See above.

**Describe the solution you'd like**
`sfdx force:source:retrieve -m 'ApexClass:SomePrefix_*'` should put in the `classes` folder only the classes that begin with `SomePrefix_`.

**Describe alternatives you've considered**
The one alternative that I can think for a wildcard retrieval would be to manually delete any files that are not in scope after the retrieval.
The alternative to a deployment with a wildcard (if it is not viable to delete files upon retrieval) is to list all the necessary classes one by one.

**Additional context**
Not applicable.

**Comment by chrisyorkston** [comment]
Is there any update on this? It would be a great feature.

**Reply by mshanemc** [reply]
in progress

**Comment by WillieRuemmele** [comment]
https://github.com/forcedotcom/cli/issues/857

**Comment by mshanemc** [comment]
delivered in sf v2.4.8
https://github.com/forcedotcom/cli/blob/main/releasenotes/README.md#248-aug-16-2023-stable-rc
